### PR TITLE
Propagate trust bundles in the data plane contract

### PR DESCRIPTION
**Pull request by pierDipi**
This is the follow up to https://github.com/knative-extensions/eventing-kafka-broker/pull/3594 for the control plane reconcilers.

- Refactor prober's clients to use `DialTLSContext` to dynamically pass TLS config based on the current values of the trust bundles + CACerts field

Fixes #

## Proposed Changes

- Trust-manager integration
- Allow administrators to add CA trust bundle to all Eventing Kafka components

**Release Note**

```release-note
Add trust-manager integration, Knative Eventing Kafka components will add to the trusted Certificate Authorities (CAs) any PEM-encoded CA certificates in any `ConfigMap` in the `knative-eventing` namespace labeled with `networking.knative.dev/trust-bundle=true`
```

**Docs**

### CODE REVIEW

**Comment by pierDipi** [comment]
@Cali0707 I think I'm close to have fixed all the tests, can you give it a early review and I'll take a look tomorrow ?

**Review by Cali0707 — COMMENTED** [review]
Just a few questions, but it generally makes sense!

File: control-plane/pkg/core/config/utils.go [hunk 1]
```diff
@@ -102,6 +105,24 @@ func EgressConfigFromDelivery(
 	return egressConfig, nil
 }
 
+func TrustBundles(lister corev1listers.ConfigMapNamespaceLister) ([]string, error) {
+	cms, err := lister.List(eventingtls.TrustBundleSelector)
```
[the remark this answers is not in the document]
Comment (pierDipi) [reply control-plane/pkg/core/config/utils.go:109]: on `List` not found is only returned when the type is not registered but for ConfigMaps is not possible as it's a core type

File: test/rekt/features/kafka_source.go [hunk 2]
```diff
@@ -539,6 +545,107 @@ func KafkaSourceTLS(kafkaSource, kafkaSink, topic string) *feature.Feature {
...
+		eventshub.IssuerRef(eventingtlstesting.IssuerKind, eventingtlstesting.IssuerName),
+	))
+
+	f.Setup("install kafka source", func(ctx context.Context, t feature.T) {
+		d := &duckv1.Destination{
+			URI: &apis.URL{
+				Scheme: "https", // Force using https
+				Host:   network.GetServiceHostname(receiver, environment.FromContext(ctx).Namespace()),
+			},
+			CACerts: nil, // CA certs are in the trust-bundle
```
[the remark this answers is not in the document]
Comment (pierDipi) [reply test/rekt/features/kafka_source.go:624]: Yes, there is one created as part of the TLS installation for eventing core

File: control-plane/pkg/core/config/utils.go [hunk 3]
```diff
@@ -102,6 +105,24 @@ func EgressConfigFromDelivery(
...
+	if err != nil {
+		return nil, fmt.Errorf("failed to list trust bundles ConfigMaps: %w", err)
+	}
+
+	cp := x509.NewCertPool()
+	trustBundles := make([]string, 0, len(cms))
+	for _, cm := range cms {
+		for _, v := range cm.Data {
+			if ok := cp.AppendCertsFromPEM([]byte(v)); ok {
+				trustBundles = append(trustBundles, v)
```
[the remark this answers is not in the document]
Comment (pierDipi) [reply control-plane/pkg/core/config/utils.go:119]: Yes, I will add a comment to clarify

**Pull request merged**